gems/smithy-client/lib/smithy-client/signers/http_basic.rb

```diff
@@ -7,11 +7,11 @@ module Client
     module Signers
       # A signer that signs requests using the HTTP Basic Auth scheme.
       class HttpBasic < Signer
-        def sign(request:, identity:, **_options)
+        def sign(context)
           # TODO: does not handle realm or other properties
-          identity_string = "#{identity.username}:#{identity.password}"
+          identity_string = "#{context[:auth].identity.username}:#{context[:auth].identity.password}"
```

Comment:
This is fine, but you can store context[:auth] since it's used twice. Hash access is always slower than local variable.